TTS/tts/utils/managers.py

```diff
@@ -0,0 +1,279 @@
+import json
+import random
+from typing import Any, Dict, List, Union
+
+import fsspec
+import numpy as np
+import torch
+
+from TTS.config import load_config
+from TTS.encoder.utils.generic_utils import setup_encoder_model
+from TTS.utils.audio import AudioProcessor
+
+
+class BaseIDManager:
+    """ Base `ID` Manager class. Every new `ID` manager must inherit this.
+    It defines common `ID` manager specific functions.
+    """
+    def __init__(
+        self,
+        id_file_path: str = ""
+    ):
+        self.ids = {}
+
+        if id_file_path:
+            self.set_ids_from_file(id_file_path)
+
+    @staticmethod
+    def _load_json(json_file_path: str) -> Dict:
+        with fsspec.open(json_file_path, "r") as f:
+            return json.load(f)
+
+    @staticmethod
+    def _save_json(json_file_path: str, data: dict) -> None:
+        with fsspec.open(json_file_path, "w") as f:
+            json.dump(data, f, indent=4)
+
+
+    def set_ids_from_data(self, items: List) -> None:
```

Comment:
arg type is not List should be Dict now. Also we could specify the key that it needs to parse from the dict.

```python
    def set_ids_from_data(self, items: Dict, parse_key: str) -> None:
```

Reply:
In fact, it is a List, not a Dict. We provide a List of Dicts.  The function parse_ids_from_data is defined in each Manager. I did it because this function can be different depending on the Manager. But yeah providing the maybe is better.

Reply:
Done